File: py/11723.py
# import sys
# input = sys.stdin.readline

# ret = 0
# for _ in range(int(input())):
#     q = list(input().split())

#     if q[0] == "add":
#         ret |= (1 << int(q[1]))
#     elif q[0] == "remove":
#         ret &= ~(1 << int(q[1]))
#     elif q[0] == "check":
#         check = ret & (1 << int(q[1]))
#         if check:
#             print(1)
#         else:
#             print(0)
#     elif q[0] == "toggle":
#         ret ^= (1 << int(q[1]))
#     elif q[0] == "all":
#         ret = (1 << 21) - 1
#     elif q[0] == "empty":
#         ret = 0


import sys
input = sys.stdin.readline

answer = 0

m = int(input())
for _ in range(m):
    row = list(input().split()) + [0]
    cmd, num = row[0], row[1]
    num = 1 << int(num)

    if cmd == "add":
        answer |= num
        
    elif cmd == "remove":
        answer &= ~num

    elif cmd == "check":
        if answer & num:
            print(1)
        else:
            print(0)

    elif cmd == "toggle":
        answer ^= num

    elif cmd == "all":
        answer = (1 << 21) - 1

    elif cmd == "empty":
        answer = 0

File: cp/11723.cpp
#include <bits/stdc++.h>
using namespace std;

int m, a, ret;
string s;

int main() {
    ios_base::sync_with_stdio(0);
    cin.tie(0); cout.tie(0);

    cin >> m;

    for(int i = 0; i < m; i++) {
        cin >> s;
        if (s == "add") {
            cin >> a;
            ret |= (1 << a);
        }
        else if (s == "remove") {
            cin >> a;
            ret &= ~(1 << a);
        }
        else if (s == "check") {
            cin >> a;
            if (ret & (1 << a)) {
                cout << 1 << "\n";
            }
            else {
                cout << 0 << "\n";
            }
        }
        else if (s == "toggle") {
            cin >> a;
            ret ^= (1 << a);
        }
        else if (s == "all") {
            ret = (1 << 21) - 1;
        }
        else if (s == "empty") {
            ret = 0;
        }
    }

    return 0;
}